iceoryx_posh/testing/roudi_environment/runtime_test_interface.cpp

```diff
@@ -112,7 +112,7 @@ PoshRuntime& RuntimeTestInterface::runtimeFactoryGetInstance(cxx::optional<const
     }
     else
     {
-        auto runtimeImpl = new runtime::PoshRuntime(name, false);
+        auto runtimeImpl = new runtime::PoshRuntimeImpl(name, false);
```

Comment:
I know that I was the one that introduced the bool `doMapSharedMemoryIntoThread` but it does not stop me to criticize it ;). Since you touched the whole `posh_runtime` could you please introduce an enum class called `MapSharedMemoryIntoThread` (or maybe you find a better name) with the two values `Enable`, `Disable` for better readability.

Reply:
Oh my gosh, I created a monster :smile: 
How about
```cpp
enum class RuntimeLocation
{
    SEPARATE_PROCESS_FROM_ROUDI,
    SAME_PROCESS_LIKE_ROUDI,
};
```
I like to give the class some context to do the right thing instead of telling what to do. This makes it easier to refactor and extend the class